pkg/steps/multi_stage/gen_test.go

```diff
@@ -379,3 +380,74 @@ func TestGetClusterClaimPodParams(t *testing.T) {
 		})
 	}
 }
+
+func TestSetSecurityContexts(t *testing.T) {
+	for _, tc := range []struct {
+		name, root string
+		pod        coreapi.Pod
+		expected   sets.String
+	}{{
+		name: "empty",
+	}, {
+		name: "no match",
+		pod: coreapi.Pod{
+			Spec: coreapi.PodSpec{
+				InitContainers: []coreapi.Container{{Name: "i0"}, {Name: "i1"}},
+				Containers:     []coreapi.Container{{Name: "c0"}, {Name: "c1"}},
+			},
+		},
+	}, {
+		name: "match",
+		pod: coreapi.Pod{
+			Spec: coreapi.PodSpec{
+				InitContainers: []coreapi.Container{{Name: "i0"}, {Name: "i1"}},
+				Containers:     []coreapi.Container{{Name: "c0"}, {Name: "c1"}},
+			},
+		},
+		root: "c1",
+	}} {
+		t.Run(tc.name, func(t *testing.T) {
+			const uid = 1007160000
+			var capabilities coreapi.Capabilities
+			var seLinuxOpts coreapi.SELinuxOptions
+			pod := &tc.pod
+			setSecurityContexts(pod, tc.root, uid, &capabilities, &seLinuxOpts)
+			f := func(c *coreapi.Container) {
```

Comment:
```suggestion
			checkSecurityContext := func(c *coreapi.Container) {
```

Reply:
nit: I would defined `expected:=&Pod{...}` and then do the assertions on the whole pod (instead of checking some fields of the pod).

Reply:
Changed to use fixtures.